### PR TITLE
Add example for exec with `tty` enabled.

### PR DESCRIPTION
**Pull request by xychu**
Including resize pty via `RESIZE_CHANNEL`.

### CODE REVIEW

File: examples/exec_tty.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
+# This example used code(the `Interceptor` related part) from
+# http://sqizit.bartletts.id.au/2011/02/14/pseudo-terminals-in-python/
+# which is licensed under the MIT license:
+# Copyright (c) 2011 Joshua D. Bartlett
```
Comment (roycaihw) [inline examples/exec_tty.py]: Sorry that I couldn't get to this earlier. I'm concerned with the MIT licensing and worried if having another license violates our [CLA](https://github.com/cncf/cla/). Did a quick search in our code base and I didn't see any source code under MIT license in client-python and kubernetes/kubernetes (third party vendors don't apply). 

@yliaog @mbohlool Do we have existing best-practice / SIG in community that we can consult with?

I'd be happy to review & merge this PR after we sort out the license question.

Comment (yliaog) [reply examples/exec_tty.py]: https://www.cncf.io/blog/2017/02/01/cncf-recommends-aslv2/

CNCF recommends Apache License. Not sure if MIT License is ok though.

**Comment by adubkov** [comment]
Sorry for interfere here but I have a related problem, maybe you can point me to right direction.

I'm trying to add similar thing like `exec -ti` in my app and i'm using this way to interact with tty using asyncio module:

```
        packed = fcntl.ioctl(sys.stdout.fileno(),
                             termios.TIOCGWINSZ,
                             struct.pack('HHHH', 0, 0, 0, 0))
        rows, cols, h_pixels, v_pixels = struct.unpack('HHHH', packed)
        ws.write_channel(4, json.dumps({"Height": rows, "Width": cols}))

        async def forward_channel_to_stream(in_stream, out_stream, channel):
            while True:
                data = await loop.run_in_executor(None, in_stream.read_channel, channel)
                await loop.run_in_executor(None, out_stream.write, data)

        async def forward_stream_to_channel(in_stream, out_stream, channel):
            while True:
                line = await loop.run_in_executor(None, in_stream.readline)
                await loop.run_in_executor(None, out_stream.write_channel, channel, line)

        stdout = forward_channel_to_stream(ws, sys.stdout, sys.stdout.fileno())
        stderr = forward_channel_to_stream(ws, sys.stderr, sys.stderr.fileno())
        stdin = forward_stream_to_channel(sys.stdin, ws, sys.stdin.fileno())
        loop = asyncio.get_event_loop()
        loop.run_until_complete(asyncio.gather(stdin, stdout, stderr))
        loop.close()
```

It works ok, but i have one problem - when i'm typing any command, lets say `ls` and press enter it print some extra codes like this:
```
/ # ls
^[[34;5Rbashrc              datadir             etc                 lib                 media               proc                sbin                start_bncr.sh       start_full.sh       thundertool         txdb
bin                 dev                 home                lib64               mnt                 root                srv                 start_cdnserver.sh  sys                 tini                usr
config              entrypoint.sh       keystore            logs                opt                 run                 start_accel.sh      start_comm.sh       thunder             tmp                 var
/ # ^[[34;5R
```

It looks like color codes or caret move, not sure why it doesn't convert to corresponding codes in my local bash.

I would appreciate any suggestions/

**Pull request closed without merging**